client/src/redux/prop-types.ts

```diff
@@ -288,6 +295,21 @@ export type ClaimedCertifications = {
   isMachineLearningPyCertV7: boolean;
 };
 
+export type SavedChallenges = {
+  id: string;
+  challengeFiles: SavedChallengeFiles;
+};
+
+export type SavedChallengeFile = {
```

Comment:
I assume once the challengeFile is fetched from the DB, the `lastSavedDate` would be there too.